app/abilities/ability.rb

```diff
@@ -43,7 +43,7 @@ def initialize(user)
         can [:new, :show, :create, :destroy], :oauth2_authorization
         can [:edit, :update, :destroy], :account
         can [:show], :dashboard
-        can [:new, :create, :edit, :update, :comment, :subscribe, :unsubscribe], DiaryEntry
+        can [:new, :create, :edit, :update, :comment, :subscribe, :unsubscribe, :hide, :unhide], DiaryEntry
```

Comment:
Is there a way to check the user here like this?

    can [:hide, :unhide], DiaryEntry, :user => user

Reply:
For example, you can put this in `hide` method:

    authorize! :hide, entry

In theory there's [`load_and_authorize_resource`](https://github.com/CanCanCommunity/cancancan/blob/develop/docs/controller_helpers.md#authorize_resource-load_resource-load_and_authorize_resource) helper.

Reply:
Sure there is a whole discussion at https://github.com/CanCanCommunity/cancancan/blob/develop/docs/fetching_records.md which suggests it can work but I haven't managed to make any sense of it yet and we're not using it anywhere else so I think we should be consistent here and look at that separately?

Reply:
I agree with both of you - there's features of CanCanCan that we can use to simplify this, but it's also something that we can tackle separately. Anton's example is correct, we can build permissions within the ability file like

```
if user
  can [:hide], DiaryEntry, :user => user # can only hide their own entries
 ....
if user.moderator?
  can [:hide], DiaryEntry # can hide all entries
```

We can then go back to using the `can :hide, diary_entry` instead of replacing that with inline permission checks like `if current_user && (current_user == diary_entry.user || current_user.moderator?)` which is logic that should be in the abilities system.

I'm happy to have this as a followup task though, particularly since merging this first will make the followup easier due to the additional tests.